bin/ember

```diff
@@ -27,5 +27,8 @@ resolve('ember-cli', {
     cliArgs: process.argv.slice(2),
     inputStream: process.stdin,
     outputStream: process.stdout
-  }).then(exit);
+  }).then(function(result) {
+    var exitCode = typeof result === 'object' ? result.exitCode : result;
```

Comment:
> I was too, but providing access to the ui for testing is definitely worth it IMO. This change essentially makes returning an object possible while still returning the exitCode.

I believe this is in-fact worth it. I just wanted to be confident that this was thoroughly taken into account. I am happy with this.